xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/ControlButton.h

```diff
@@ -0,0 +1,175 @@
+#pragma once
+/*
+ *      Copyright (C) 2005-2017 Team KODI
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with KODI; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "../AddonBase.h"
+#include "Window.h"
+
+namespace kodi
+{
+namespace gui
+{
+
+  //============================================================================
+  ///
+  /// \defgroup cpp_kodi_gui_CControlButton Control Button
+  /// \ingroup cpp_kodi_gui
+  /// @brief \cpp_class{ kodi::gui::CControlButton }
+  /// **Standard push button control for window**
+  ///
+  /// The  button  control  is used for creating push buttons  in Kodi.  You can
+  /// choose the position,  size,  and look of the button,  as well as  choosing
+  /// what action(s) should be performed when pushed.
+  ///
+  /// It has the header \ref ControlButton.h "#include <kodi/gui/ControlButton.h>"
+  /// be included to enjoy it.
+  ///
+  /// Here you find the needed skin part for a \ref skin_Button_control "button control"
+  ///
+  /// @note The call of the control is  only  possible  from  the  corresponding
+  /// window as its class and identification number is required.
+  ///
+  class CControlButton : public CAddonGUIControlBase
```

Comment:
I would not use "CControl" prefix for all classes modelling GUI controls. This is what namespaces are for. 

I would use just "CButton" here. For me it is quite obvious  that a class contained in namespace "kodi::gui" named "CButton" is a GUI control. Even better, you could put all GUI controls in namespace kodi::gui::control.